### PR TITLE
added dac support

### PR DESCRIPTION
**Pull request by hardfau18**
solves #9 
shamelessly copied and modified from stm32f4xx-hal
in future  will add alignment and different bit supports

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -25,7 +25,7 @@ micromath = "1.0.0"
 synopsys-usb-otg = { version = "0.2.3", features = ["cortex-m"], optional = true }
 stm32-fmc = { version = "0.2.0", features = ["sdram"], optional = true }
 rand_core = "0.5.1"
-bxcan = "0.4"
```
Comment (hardfau18) [inline Cargo.toml:28]: defmt 2.0 requires this

**Review by hannobraun — COMMENTED** [review]
Thank you, @hardfau18!

Some comments:
- I don't understand the reasoning for having a public `dac` function _and_ the `DacExt` trait. Seems redundant, but I guess it's fine. We can always clean it up later, whenever someone has the time and inclination to do so.
- I'm not a fan of the extension trait style (`DacExt`), as I think it introduces complexity (user has to import the trait) for no good reason. I prefer peripheral structs with a simple `new` constructor instead. But again, it's fine. Can always be cleaned up later.

I assume you tested this successfully? If so, I'd be happy to merge it.

**Comment by hardfau18** [comment]
tested it.
About changes requested I am working on rewriting everything. but may take some time

**Comment by hannobraun** [comment]
Great, thanks again!

**Pull request merged**